docs/src/man/basics.md

```diff
@@ -0,0 +1,2170 @@
+──────┼─────────────────────────────────────────────────────────────────────────
+    1 │     0     67  male        2  own      NA               little          ⋯
+    2 │     1     22  female      2  own      little           moderate
+    3 │     2     49  male        1  own      little           NA
+    4 │     3     45  male        2  free     little           little
+    5 │     4     53  male        2  free     little           little          ⋯
+    6 │     5     35  male        1  free     NA               NA
+    7 │     6     53  male        2  own      quite rich       NA
+    8 │     7     35  male        3  rent     little           moderate
+  ⋮   │   ⋮      ⋮      ⋮       ⋮       ⋮            ⋮                ⋮        ⋱
+  994 │   993     30  male        3  own      little           little          ⋯
+  995 │   994     50  male        2  own      NA               NA
+  996 │   995     31  female      1  own      little           NA
+  997 │   996     40  male        3  own      little           little
+  998 │   997     38  male        2  own      little           NA              ⋯
+  999 │   998     23  male        2  free     little           little
+ 1000 │   999     27  male        2  own      moderate         moderate
+                                                  4 columns and 985 rows omitted
+```
+
+We will get the vector stored in our DataFrame without copying it:
+
+```jldoctest dataframe
+julia> german.Age 
+1000-element Vector{Int64}:
+ 67
+ 22
+ 49
+ 45
+ 53
+ 35
+ 53
+ 35
+ 61
+ 28
+  ⋮
+ 34
+ 23
+ 30
+ 50
+ 31
+ 40
+ 38
+ 23
+ 27
+```
+
+We will get the vector stored in our data frame without copying it using String:
+
+```jldoctest dataframe
+julia> german."Sex" 
+1000-element PooledArrays.PooledVector{String, UInt32, Vector{UInt32}}:
+ "male"
+ "female"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ ⋮
+ "male"
+ "male"
+ "male"
+ "male"
+ "female"
+ "male"
+ "male"
+ "male"
+ "male"
+
+julia> german[:, 3] # note that this creates a copy
+1000-element PooledArrays.PooledVector{String, UInt32, Vector{UInt32}}:
+ "male"
+ "female"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ ⋮
+ "male"
+ "male"
+ "male"
+ "male"
+ "female"
+ "male"
+ "male"
+ "male"
+ "male"
+```
+
+To get two columns as a `DataFrame`, we can index as follows:
+
+```jldoctest dataframe
+julia> german[:, 1:2]
+1000×2 DataFrame
+  Row │ id     Age
+      │ Int64  Int64
+──────┼──────────────
+    1 │     0     67
+    2 │     1     22
+    3 │     2     49
+    4 │     3     45
+    5 │     4     53
+    6 │     5     35
+    7 │     6     53
+    8 │     7     35
+  ⋮   │   ⋮      ⋮
+  994 │   993     30
+  995 │   994     50
+  996 │   995     31
+  997 │   996     40
+  998 │   997     38
+  999 │   998     23
+ 1000 │   999     27
+     985 rows omitted
+
+julia> german[:, [3]] # here we have created a single column data frame
```

Comment:
please do not use code comments to write explanations - transform it into markdown.